tests/unit/test_response_generators.py

```diff
@@ -0,0 +1,297 @@
+        self.assertIsNotNone(alignment)
+        self.assertIsNotNone(nextPageToken)
+        alignment, nextPageToken = next(iterator)
+        self.assertIsNotNone(alignment)
+        self.assertIsNone(nextPageToken)
+        self.assertIsNone(next(iterator, None))
+
+    def _initReadGroup(self, numAlignments):
+        readGroup = MockReadGroup(self.readGroupId, numAlignments)
+        self.backend._readGroupIdMap = {self.readGroupId: readGroup}
+        self.request.readGroupIds = [self.readGroupId]
+
+
+class TestReadsGeneratorMethods(unittest.TestCase):
+    """
+    Test the helper methods for readsGenerator
+    """
+    def setUp(self):
+        self.request = protocol.GASearchReadsRequest()
+        self.request.start = 3
+        self.request.pageToken = "notNone"
+        self.backend = backend.AbstractBackend()
+        self.alignmentPosition = 5
+        self.startPosition = 1
+        self.equalPositionsToSkip = 1
+        self.variant = protocol.GAVariant()
+        self.variant.start = 4
+        self.variant.end = 6
+
+    def testGetReadAlignmentStart(self):
+        alignment = generateReadAlignment(self.alignmentPosition)
+        position = backend._getReadAlignmentStart(alignment)
+        self.assertEqual(position, self.alignmentPosition)
+
+    def testGetReadAlignmentEnd(self):
+        alignment = generateReadAlignment(self.alignmentPosition)
+        endPosition = backend._getReadAlignmentEnd(alignment)
+        self.assertEqual(
+            self.alignmentPosition + 3, endPosition)
+
+    def testGetVariantStart(self):
+        result = backend._getVariantStart(self.variant)
+        self.assertEqual(self.variant.start, result)
+
+    def testGetVariantEnd(self):
+        result = backend._getVariantEnd(self.variant)
+        self.assertEqual(self.variant.end, result)
+
+    def testGetCounters(self):
+        # with pageToken == None
+        startPos = 9
+        self.request.start = startPos
+        self.request.pageToken = None
+        startPosition, equalPositionsToSkip = self.backend._getCounters(
+            self.request)
+        self.assertEqual(startPosition, startPos)
+        self.assertEqual(equalPositionsToSkip, 0)
+
+        # with pageToken != None
+        self.request.pageToken = "1:2"
+        startPosition, equalPositionsToSkip = self.backend._getCounters(
+            self.request)
+        self.assertEqual(startPosition, 1)
+        self.assertEqual(equalPositionsToSkip, 2)
+
+    def testPositionIterator(self):
+        def getIterator():
+            # skipped because position < startPosition
+            yield generateReadAlignment(0)
+            # skipped because endPosition < request.start
+            yield generateReadAlignment(1, 'a')
+            # skipped because equalPositionsSkipped
+            yield generateReadAlignment(1, 'abcdefg')
+            # this one should be returned
+            yield generateReadAlignment(1, 'abcdefghijklmnop')
```

Comment:
Can you put this string into a variable for testing against? It's too hard to read it to make sure it's the same string below.